tests/user_commands/tests.py

```diff
@@ -468,6 +468,18 @@ def test_skip_checks(self):
         self.assertNoOutput(err)
         self.assertEqual(out.strip(), "Set foo")
 
+    def test_subparser_error_formatting(self):
+        self.write_settings("settings.py", apps=["user_commands"])
+        out, err = self.run_manage(["subparser", "foo", "twelve"])
+        self.assertNoOutput(out)
+        self.assertOutput(err, "argument bar: invalid int value: 'twelve'")
+
+    def test_subparser_nondjango_error_formatting(self):
+        self.write_settings("settings.py", apps=["user_commands"])
+        out, err = self.run_manage(["subparser_vanilla", "foo", "seven"])
+        self.assertNoOutput(out)
+        self.assertOutput(err, "argument bar: invalid int value: 'seven'")
```

Comment:
Both tests work after removing `CommandParser.add_subparsers()` :thinking: Can you add a regression test?

Reply:
Right you are, the formatted error message the test was checking for also appeared in the stack trace. I've changed the assertions to be stricter, the first test now fails on `main`. The second test is there to check no regression in handling explicitly vanilla subparsers.